tests/unittests/ExecutorTest.cpp

```diff
@@ -0,0 +1,527 @@
+  auto betaInputContext = std::make_unique<Context>(
+      cloneContext(ctx_, /*whitelist=*/{betaIn.get()}));
+  auto betaOutputContext = std::make_unique<Context>(
+      cloneContext(ctx_, /*whitelist=*/{betaOut.get()}));
+  auto gammaInputContext = std::make_unique<Context>(
+      cloneContext(ctx_, /*whitelist=*/{alphaOut.get(), betaOut.get()}));
+  auto gammaOutputContext = std::make_unique<Context>(
+      cloneContext(ctx_, /*whitelist=*/{deltaIn.get(), epsIn.get()}));
+  auto deltaInputContext = std::make_unique<Context>(
+      cloneContext(ctx_, /*whitelist=*/{deltaIn.get()}));
+  auto deltaOutputContext = std::make_unique<Context>(
+      cloneContext(ctx_, /*whitelist=*/{deltaOut.get()}));
+  auto epsInputContext = std::make_unique<Context>(
+      cloneContext(ctx_, /*whitelist=*/{epsIn.get()}));
+  auto epsOutputContext = std::make_unique<Context>(
+      cloneContext(ctx_, /*whitelist=*/{epsOut.get()}));
+
+  // Create the Context that Executor::run() is expected to return.
+  auto refResultContext =
+      cloneContext(ctx_, /*whitelist=*/{deltaOut.get(), epsOut.get()});
+
+  // Register the expected input Contexts and desired output Contexts for
+  // all nodes with the TestDeviceManager.
+  deviceManager->registerResult("alpha", 0, ResultCode::Executed,
+                                std::move(alphaInputContext),
+                                std::move(alphaOutputContext));
+  deviceManager->registerResult("beta", 0, ResultCode::Executed,
+                                std::move(betaInputContext),
+                                std::move(betaOutputContext));
+  deviceManager->registerResult("gamma", 0, ResultCode::Executed,
+                                std::move(gammaInputContext),
+                                std::move(gammaOutputContext));
+  deviceManager->registerResult("delta", 0, ResultCode::Executed,
+                                std::move(deltaInputContext),
+                                std::move(deltaOutputContext));
+  deviceManager->registerResult("eps", 0, ResultCode::Executed,
+                                std::move(epsInputContext),
+                                std::move(epsOutputContext));
+
+  // Get the root node of the DAG constructed earlier.
+  DAGNode *root = dag.getRoot();
+
+  // Variables for storing runId and resultCode actually returned by
+  // Executor::run() via its callback.
+  RunIdentifierTy resultRunId;
+  ResultCode resultCode;
+  std::unique_ptr<Context> resultContext;
+
+  // Call Executor::run().
+  E_->run(root, std::move(inputContext), testRunId,
+          [&resultRunId, &resultCode,
+           &resultContext](RunIdentifierTy runId, ResultCode code,
+                           std::unique_ptr<Context> context) {
+            resultRunId = runId;
+            resultCode = code;
+            resultContext = std::move(context);
+          });
+
+  std::this_thread::sleep_for(std::chrono::seconds(2));
```

Comment:
does this mean that our unittests sleep for 2 seconds?   Every unit test? I don't think that this scales with the number of tests. What happens when we get to 100 tests? We will wait 200 seconds?